bin/mage

```diff
@@ -1,55 +1,196 @@
-#! /usr/bin/env bb
-
-;; Getting env: bb: No such file or directory ?
-;; Install bb with: brew install borkdude/brew/babashka
-
-;; you may want to set an alias for mage:
-;; alias mage='cd ~/your/repo/metabase && ./bin/mage'
-
-(ns mage
-  (:require
-   [babashka.tasks :as bt]
-   [mage.color :as c]
-   [mage.shell :as sh]
-   [mage.util :as u]))
-
-(set! *warn-on-reflection* true)
-
-(defn lolcat [f]
-  (if (try (u/sh "command -v lolcat")
-           (catch Exception _ false))
-    (bt/shell (str "lolcat " f))
-    (println (slurp f))))
-
-(defn invalid-task? []
-  (let [task-name (first *command-line-args*)]
-    (when-not (contains? (set (u/all-bb-tasks-list)) task-name)
-      (println (c/red (str "Unknown task: " task-name)))
-      true)))
-
-(defn- print-help []
-  (do
-    (lolcat "./mage/resource.txt")
-    (flush)
-    (println (c/bold " ✨ Metabase Automation Genius Engine ✨"))
-    (println "")
-    (println (u/sh "bb tasks"))))
-
-(defn -main [& _]
-  (cond
-    ;; help
-    (or
-     (nil? *command-line-args*)
-     (= *command-line-args* ["-h"])
-     (= *command-line-args* ["--help"]))
-    (print-help)
-
-    ;; errors
-    (invalid-task?)
-    (do (print-help) (System/exit 1))
-
-    :else
-    (apply sh/sh (into ["bb"] *command-line-args*))))
-
-(when (= *file* (System/getProperty "babashka.file"))
-  (-main))
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Debug function to see file content
+debug_file_content() {
+  if [ -n "${MAGE_DEBUG:-}" ]; then
+    local file="$1"
+    echo "Content of $file:" >&2
+    cat "$file" >&2
+    echo "" >&2
+  fi
+}
+
+# Get Babashka version from bb.edn
+get_bb_version() {
+  local edn_file="${BB_EDN_PATH:-"../bb.edn"}"
+  local default_version="1.3.186"  # Fallback version
+
+  if [ ! -f "$edn_file" ]; then
+    log_warning "bb.edn not found at ${edn_file}, using default version: ${default_version}"
+    echo "$default_version"
+    return
+  fi
+
+  # Debug: print file content when in debug mode
+  # debug_file_content "$edn_file"
+
+  # Simple approach for the specific format :min-bb-version "x.y.z"
+  bb_version=$(grep -o ':min-bb-version "[0-9]\+\.[0-9]\+\.[0-9]\+"' "$edn_file" | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+')
```

Comment:
idk if you'd like the awk version?

```
@imai ~/dev/work/metabase> awk '/:min-bb-version/ { print $2 }' bb.edn
"1.12.196"
```

This code is very diligent on version format, but I think it's better to be a bit less strict and just go with whatever is in `:min-bb-version`? We could even test that if you feel like it 😁